### DISCUSSION TITLE
Digital Inputs

### DISCUSSION THREAD

**Discussion by mcinquino** [fra589/grbl-Mega-5X#280] [Q&A] [answered]
Hello,

I see that in the pinout diagram there are 4 digital inputs on pins 27,29,31 and 33 that are shown. I see that we have M codes -  M62, M63, M64, M65 to set the outputs that are shown above these inputs. In linuxCNC we would use M66 to wait on an input. I don't see that M66 is implemented in the list of G codes. Is there a way to use these inputs in G code? 

![image](https://user-images.githubusercontent.com/15634931/181789795-893071d6-4987-477a-b703-5de60559e053.png)

Thanks,
Mike

**Comment by fra589** [comment, marked as answer]
Hi @mcinquino,  
  
M66 is not yet implemented since nobody have asked it for now :)  
Normally, it will be not so difficult to add M66 to the GCode interpreter...
  
I transfer this discuss to new enhancement issue.

@++;  
Gauthier.

**Comment by mcinquino** [comment]
Hi, Thank You! I will keep checking back to see if this gets added and I
can do any testing that may be needed. Looking forward to using this
feature.
